presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/TypeValidator.java

```diff
@@ -170,10 +173,33 @@ private void checkFunctionSignature(Map<VariableReferenceExpression, Aggregation
         private void checkAggregation(Map<VariableReferenceExpression, Aggregation> aggregations)
         {
             for (Map.Entry<VariableReferenceExpression, Aggregation> entry : aggregations.entrySet()) {
+                VariableReferenceExpression variable = entry.getKey();
+                Aggregation aggregation = entry.getValue();
+                FunctionMetadata functionMetadata = metadata.getFunctionManager().getFunctionMetadata(aggregation.getFunctionHandle());
                 verifyTypeSignature(
-                        entry.getKey(),
-                        metadata.getFunctionManager().getFunctionMetadata(entry.getValue().getFunctionHandle()).getReturnType());
-                // TODO check if the argument type agrees with function handle (will be added once Aggregation is using CallExpression).
+                        variable,
+                        functionMetadata.getReturnType());
+                verifyTypeSignature(
+                        variable,
+                        aggregation.getCall().getType().getTypeSignature());
+                int argumentSize = aggregation.getArguments().size();
```

Comment:
Curious: Do we have corresponding check in the `Expression` world? Or they are newly enabled with this `RowExpression` migration?

Reply:
Yes, we previously cannot check argument types because we simplify don't know them.